tests/net/lib/coap/src/main.c

```diff
@@ -932,7 +932,7 @@ static int test_block_size(void)
 	struct net_pkt *pkt = NULL;
 	struct net_buf *frag;
 	const char token[] = "token";
-	u8_t payload = 0xFE;
+	u8_t payload[32] = { 0 };
```

Comment:
How about "u8_t payload[coap_block_size_to_bytes(COAP_BLOCK_32)] = { 0 };" ?

Reply:
Otherwise, I am planning to add some test cases with malformed packets. I will modify this test case a little bit.

Reply:
Although `coap_block_size_to_bytes()` is a `static inline` function, the `inline`  qualifier is just a suggestion; the compiler might ignore it.  If it's ignored, this will be a VLA, which is unspecified by the standard; GCC will most likely use the builtin for `alloca()`, which is fine for a size this small.

However, since the size is not going to change, and this is testing code, I don't see much of a problem in leaving the hardcoded constant in there.